### PR TITLE
Checkbox: Refactor indeterminate styles for size reduction

### LINKED ISSUE #10150: Fabric 7.23 is adding ~1.3kB to Checkbox
Pr responsible for size increase: https://github.com/OfficeDev/office-ui-fabric-react/pull/9815
This seems to be a big increase for an extra feature.

### Priorities and help requested:

Are you willing to submit a PR to fix? No

Requested priority: High

Products/sites affected: OutlookWeb fabric updates

### PR DESCRIPTION
**Pull request by maxwellred**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #10150 
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Refactoring some of the styles to reduce component size

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10170)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts [hunk 1]
```diff
@@ -20,52 +20,43 @@ export const getStyles = (props: ICheckboxStyleProps): ICheckboxStyles => {
...
+    borderHovered: semanticColors.inputBorderHovered,
+    borderDisabled: semanticColors.disabledBodySubtext,
+    backgroundChecked: semanticColors.inputBackgroundChecked,
+    backgroundCheckedHovered: semanticColors.inputBackgroundCheckedHovered,
+    borderCheckedHovered: semanticColors.inputBackgroundCheckedHovered,
+    backgroundDisabledChecked: semanticColors.disabledBodySubtext,
+    text: semanticColors.bodyText,
+    textHovered: semanticColors.inputTextHovered,
+    textDisabled: semanticColors.disabledText
+  };
```
Comment (JasonGore) [inline packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts]: Hmm, size auditor results are saying size actually went up by 322 bytes. This structure may actually work against you. I suspect references to `checkboxColor.backgroundCheckedHovered` may only reduced to `c.backgroundCheckedHovered`. May be worth trying the original approach to confirm.

File: packages/office-ui-fabric-react/src/components/__snapshots__/Checkbox.Other.Example.tsx.shot [hunk 2]
```diff
@@ -1570,11 +1559,7 @@ exports[`Component Examples renders Checkbox.Other.Example.tsx correctly 1`] = `
               height: 10px;
               left: 4px;
               position: absolute;
-              right: 0px;
               top: 4px;
-              transition-duration: 200ms;
-              transition-property: border-width, border, border-color;
-              transition-timing-function: cubic-bezier(.4, 0, .23, 1);
```
Comment (JasonGore) [inline packages/office-ui-fabric-react/src/components/__snapshots__/Checkbox.Other.Example.tsx.shot]: Are these changes intentional? I had started on a PR myself, but I didn't htink there'd be any changes to actual styling in snapshots.

**Comment by JasonGore** [comment]
Thanks for the PR!

File: packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts (hunk 1 above)
Comment (maxwellred) [reply packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts]: Interesting, I calculated I removed around 700 characters. I'll revert it like you say to see if there's a difference

Comment (JasonGore) [reply packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts]: There's not always a direct correlation between what you remove and what can be minified. The object and variable names can be minified, but the key names cannot. You removed a bunch of variable names that can be minified, but then added a bunch of references to keys, which can't be minified. 😄

Comment (JasonGore) [reply packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts]: As an example:

Before: 
`checkboxBorderIndeterminateColor ` could reduce to `c`

After:
`checkboxColor.borderIndeterminate` can only reduce to `c.borderIndeterminate` because key names can't be minified

File: packages/office-ui-fabric-react/src/components/__snapshots__/Checkbox.Other.Example.tsx.shot (hunk 2 above)
Comment (maxwellred) [reply packages/office-ui-fabric-react/src/components/__snapshots__/Checkbox.Other.Example.tsx.shot]: My mistake. I commented them out looking for fat to trim and deleted them on accident before the push. I'll pop them back, thanks!

File: packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts (hunk 1 above)
Comment (maxwellred) [reply packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.styles.ts]: Oh! Great explanation, that makes 100% sense and look at that we dropped the size down -258 bytes!

**Comment by maxwellred** [comment]
@JasonGore Alright so in the indeterminateDot after the refactor there is an unnecessary {right: 0} that I removed. It causes a snapshot to be different but doesn't change anything visually.

**Comment by JasonGore** [comment]
(I also confirmed with @apneer that these results resolve #10150.)

File: packages/office-ui-fabric-react/src/components/__snapshots__/Checkbox.Other.Example.tsx.shot [hunk 3]
```diff
@@ -1117,7 +1117,6 @@ exports[`Component Examples renders Checkbox.Other.Example.tsx correctly 1`] = `
               height: 10px;
               left: 4px;
               position: absolute;
-              right: 0px;
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/__snapshots__/Checkbox.Other.Example.tsx.shot]: Looks like the primary change to the control is the removal of the right value. Is this necessary? Or should we expect the snapshot to be a zero change?

Comment (JasonGore) [reply packages/office-ui-fabric-react/src/components/__snapshots__/Checkbox.Other.Example.tsx.shot]: He commented about that [here](https://github.com/OfficeDev/office-ui-fabric-react/pull/10170#issuecomment-522155358)

**Comment by JasonGore** [comment]
@aftab-hassan I was hoping you could take a look at this. Size auditor reported a -258 byte improvement. Then a change log commit was added, and the size improvement reduced to -56 bytes. Did the change log file somehow affect the bundle size?

**Comment by aftab-hassan** [comment]
> @aftab-hassan I was hoping you could take a look at this. Size auditor reported a -258 byte improvement. Then a change log commit was added, and the size improvement reduced to -56 bytes. Did the change log file somehow affect the bundle size?

I see how the comment was edited to the new size. I am taking a look at this.

**Comment by JasonGore** [comment]
I spoke with Anna and I will merge this for now. This PR should be improving bundle size, we're just not sure how much right now. If we still need further bundle improvements we can follow up.
